R/utlede_segmentStent_variabler.R

```diff
@@ -0,0 +1,264 @@
+#' Add number of stents per procedure in AP-dataset
+#'
+#' @param df_ap Angio PCI dataset where new variable `antall_stent` should be
+#' added. Must contain the variables `AvdRESH` and `ForlopsID`,
+#' @param df_ss segment stent dataset where `antall_stent` is calculated from.
+#' Must contain the variables `AvdRESH`, `ForlopsID` and `StentType`
+#' @return data.frame with one new variable
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' AP %>% legg_til_antall_stent(., df_ss = SS)
+#' }
+legg_til_antall_stent <- function(df_ap, df_ss) {
+
+  if (!all(c("AvdRESH", "ForlopsID") %in% names(df_ap))) {
+    stop("df_apmust contain variables AvdRESH + ForlopsID")
+  }
+
+  if (!all(c("AvdRESH", "ForlopsID", "StentType") %in% names(df_ss))) {
+    stop("ss must contain variables AvdRESH + ForlopsID + StentType")
+  }
+
+  message("Legger til variabelen antall_stent...")
+
+    # In SS-dataset : Count number of non-missing entries in StentType for each
+  # procedure
+  ant_stent <- df_ss %>%
+    dplyr::select(.data$AvdRESH, .data$ForlopsID, .data$StentType) %>%
+    dplyr::arrange(., .data$AvdRESH)  %>%
+    dplyr::group_by(.data$AvdRESH) %>%
+    dplyr::count(.data$ForlopsID,
+                 wt = !is.na(.data$StentType)) %>%
+    dplyr::rename("antall_stent" = .data$n)
+
+
+  df_ap %>%
+    dplyr::left_join(.,
+                     ant_stent,
+                     by = c("AvdRESH", "ForlopsID")) %>%
+    dplyr::arrange(.data$AvdRESH, .data$ForlopsID)
+}
+
+
+
+
+
+
+#' Add variable `kar` to NORIC segment-stent-table
```

Comment:
Det er (kanskje) ikke nødvendig med en full dokumentasjon av denne funksjonen. Hvis (dokumentasjonen av) argumentene, eksempelvis `df_ss`, er det samme som funksjonen over kan det hende det går an ha felles dokumentasjon for alle funksjoner i denne fila, [eksempelvis slik det er gjort i `getData()` i `ablanor`](https://github.com/Rapporteket/ablanor/blob/main/R/getData.R).